### DISCUSSION TITLE
Workspace Idea for working with your Solid Panes

### DISCUSSION THREAD

**Discussion by SharonStrats** [solid/solidos#93] [Ideas]
My idea is to bring the panes you use the most to the forefront of your pod in a way that you can use them easily and all at once. The idea would be to follow @timbl vision of "You can do anything with anything!" On this "workspace" app there would be mini widgets of the panes we already have. They would be configurable by the user (size, perhaps color, anything we can think of) and draggable (stored in state as to where you would like them placed). The widget itself would also be configured by the app designer following standards via an interface/api that we create. I envision this interface to include a few things. What data (triples) to display to the user inside the widget, what actions to include in an action bar (a modal that appears on a right click of the data (triple)). We could use the existing preferences to store this information (might need to restructure preferences a bit to include this).

Some additions
- the data that is shown in the widget could potentially be from multiple pods, again configurable and showing a specific color based on the pod it is from. So for instance my kids could have pods and share their calendar with me, or we could share to do lists. I could then see my childs calendar items or add a todo to their list - do your homework :).
- the data (triples) themselves could be in a div that is both draggable and a drop target, when you drag one piece of data to another depending on what the app designer had as a drop target function it would do something with it.  As an example if I dragged a contact to a Friends widget it would add my contact to my friends list, if I drag a friend to a particular chat it could open up the chat or it could add the friend to the chat.

A drawing of this idea is to come.

**Comment by timbl** [comment]
This sounds like a sort of desktop/phonetop/dashboard sort of thing.   Presumably you might have one address book, two groups, and and several chats (say) ... so each item is a thing you were looking at combined with the pane you were using at the time?

**Reply by SharonStrats** [reply]
Yes exactly. But what you just said made me realize I have not considered that in each "pane mini" you would want to pick/favorite the triples you want to see or at least see at the top otherwise you might have way too much data to show and it may or may not be relevant.

I had originally thought of dashboard, but we already have a dashboard. We also have workspaces... so I'm having trouble coming up with another name for this idea. It would appear in your menu, maybe below "Show your Profile" you would have something like "My Workspace".
